wrappers/hnygrpc/grpc.go

```diff
@@ -72,16 +74,26 @@ func addFields(ctx context.Context, info *grpc.UnaryServerInfo, handler grpc.Una
 	span.AddField("handler.name", handlerName)
 	span.AddField("handler.method", info.FullMethod)
 
-	md, ok := metadata.FromIncomingContext(ctx)
+	pr, ok := peer.FromContext(ctx)
 	if ok {
-		if val, ok := md["content-type"]; ok {
-			span.AddField("request.content_type", val[0])
-		}
-		if val, ok := md[":authority"]; ok {
-			span.AddField("request.header.authority", val[0])
+		if pr.Addr != net.Addr(nil) {
+			span.AddField("request.remote_addr", pr.Addr.String())
 		}
-		if val, ok := md["user-agent"]; ok {
-			span.AddField("request.header.user_agent", val[0])
+	}
+
+	headersToFields := map[string]string{
+		"content-type":      "request.content_type",
+		":authority":        "request.header.authority",
```

Comment:
Does this want a leading : unlke all the others?

Reply:
Yup. [RFC 7540 pseudo headers](https://www.rfc-editor.org/rfc/rfc7540#section-8.1.2.3). The colon was new to me, too.

Reply:
Sweet - thanks for indulging my curiosity.